### PR TITLE
Add checking GMT Legacy/Dev Tests to the release checklist

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Related to https://github.com/GenericMappingTools/pygmt/pull/2281#issuecomment-1366768553.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
We can wait until the GMT team decides if a Mastodon account will be created (c.f. https://github.com/GenericMappingTools/pygmt/issues/2244#issuecomment-1368280782).

**Comment by seisman** [comment]
> We can wait until the GMT team decides if a Mastodon account will be created (c.f. [#2244 (comment)](https://github.com/GenericMappingTools/pygmt/issues/2244#issuecomment-1368280782)).

No progress on the Mastodon account and we're going to make a new PyGMT release. So now it's time to review/merge this PR.

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ assignees: ''
 - [ ] Wrap Y ()
 
 **Before release**:
+- [ ] Check if all tests pass in the ["GMT Legacy Tests" workflow](https://github.com/GenericMappingTools/pygmt/actions/workflows/ci_tests_legacy.yaml)
+- [ ] Check if all tests pass in the ["GMT Dev Tests" workflow](https://github.com/GenericMappingTools/pygmt/actions/workflows/ci_tests_dev.yaml)
 - [ ] Run `grep --include="*.py" -r 'remove_version="vX.Y.Z"' pygmt` from the base of the repository to check if any deprecations and related tests should be removed in this version
```
Comment (weiji14) [inline .github/ISSUE_TEMPLATE/release_checklist.md:21]: Maybe put all the 'check's under a nested list?

```suggestion
- [ ] Check to ensure that:
  - [ ] All tests pass in the ["GMT Legacy Tests" workflow](https://github.com/GenericMappingTools/pygmt/actions/workflows/ci_tests_legacy.yaml)
  - [ ] All tests pass in the ["GMT Dev Tests" workflow](https://github.com/GenericMappingTools/pygmt/actions/workflows/ci_tests_dev.yaml)
  - [ ] Deprecations and related tests are removed for this version by running `grep --include="*.py" -r 'remove_version="vX.Y.Z"' pygmt` from the base of the repository
```

**Review by weiji14 — COMMENTED** [review]
> > We can wait until the GMT team decides if a Mastodon account will be created (c.f. [#2244 (comment)](https://github.com/GenericMappingTools/pygmt/issues/2244#issuecomment-1368280782)).
> 
> No progress on the Mastodon account and we're going to make a new PyGMT release. So now it's time to review/merge this PR.

Thanks for the reminder. I've added the Mastodon account discussion to the agenda for the upcoming PyGMT Community Meeting at https://forum.generic-mapping-tools.org/t/pygmt-community-meeting-tuesday-14-march-2023-utc-1800/3734.

**Review by weiji14 — APPROVED** [review]

**Review by yvonnefroehlich — APPROVED** [review]

**Pull request merged**